Search Endpoint

HUmutOzdemir opened the pull request:
Please Make a code review and merge if there is not any problem. If there are sone things that must change please write on the linked issue about Search Endpoint.

Please control these criteria.(If you want you can add extra criteria.)
- [x] Documentation
- [x] Readability
- [x] Code works correctly or not?
- [x] Tests

oykuyilmaz1 commented:
I analyzed your code and maybe you can think of changing those:
- Your test.py does not have enough documentation I think, you have comments that clarifying the functions of your unit test but maybe you can add some definitions to that.
- in your searchEngine.py, you have a nice documentation and when I read I can see what the functions are doing clearly and it works. But you can review the error messages as follows: 
* Line 88 you have a typo --> "parameters"
* Line 100 --> "There is a problem about the search API. Please try again!!" (So that the user have more information of what went wrong)
* Line 241 --> "You have entered a wrong keyword, or selected wrong filters. Or maybe you have not logged in." (I think getparameters function also checks if the user has logged in if not please ignore it. Also you can improve the sentence ot correct it. The point is to be more specific to the user about the error.)
* Line 244 --> Maybe you have not logged in? (Same argument goes for this again.)
* Line 252 -->"There is a problem about the search API. Please try again!!" 
Other than that I think your code is readable, works correctly and well indented. Thank you for your effort.

HUmutOzdemir commented:
Your feedback seems true. I will update the code until tomorrow. Is this okey for you?

oykuyilmaz1 commented:
Sure, it is fine. Sorry for the late response.

ertbulbul commented on practice-app/platon_api/rest_api/search_engine/search_engine.py:
```diff
@@ -0,0 +1,295 @@
+
+            returns sorted version of the search result
+
+            This function sorts the output with a given sotring criteria
```
sorting misspelled

HUmutOzdemir replied:
Done

ertbulbul commented on practice-app/platon_api/rest_api/search_engine/documentation/search_engine_documentation.md:
```diff
@@ -0,0 +1,126 @@
+     |
+     |  num_of_semantically_related = 5
+     |
+     |  sotring_criteria_list = ['name_increasing', 'name_decreasing', 'surnam...
```
sorting misspelled

HUmutOzdemir replied:
Done

ertbulbul commented on practice-app/platon_api/rest_api/search_engine/documentation/search_engine_documentation.md:
```diff
@@ -0,0 +1,126 @@
+     |
+     |      returns sorted version of the search result
+     |
+     |      This function sorts the output with a given sotring criteria
```
sorting misspelled

HUmutOzdemir replied:
Done

HUmutOzdemir commented:
> I analyzed your code and maybe you can think of changing those:
> 
> * Your test.py does not have enough documentation I think, you have comments that clarifying the functions of your unit test but maybe you can add some definitions to that.
> * in your searchEngine.py, you have a nice documentation and when I read I can see what the functions are doing clearly and it works. But you can review the error messages as follows:
> 
> * Line 88 you have a typo --> "parameters"
> * Line 100 --> "There is a problem about the search API. Please try again!!" (So that the user have more information of what went wrong)
> * Line 241 --> "You have entered a wrong keyword, or selected wrong filters. Or maybe you have not logged in." (I think getparameters function also checks if the user has logged in if not please ignore it. Also you can improve the sentence ot correct it. The point is to be more specific to the user about the error.)
> * Line 244 --> Maybe you have not logged in? (Same argument goes for this again.)
> * Line 252 -->"There is a problem about the search API. Please try again!!"
>   Other than that I think your code is readable, works correctly and well indented. Thank you for your effort.

I added extra documentation to the unit tests. Also I changed the error messages that will be seen but the user. Other strings that I returned in the get_parameters functions are some messages for me actually. These messages will not be seen by the user.

HUmutOzdemir commented on practice-app/platon_api/rest_api/search_engine/search_engine.py:
```diff
@@ -0,0 +1,295 @@
+        """
+            where 'request': HTTP request that comes from the view class
+
+            returns a dictionary of the paramters if there is no problem about the paramters
```
I changed this

HUmutOzdemir commented on practice-app/platon_api/rest_api/search_engine/search_engine.py:
```diff
@@ -0,0 +1,295 @@
+            return "Wrong input. Please give an appropriate input!!"
+        if request.GET.get("sorting_criteria",False):
+            # Take sorting criteria from URL
+            sotring_criteria = request.GET.get("sorting_criteria",False).replace('"',"")
```
I changed this also

HUmutOzdemir commented on practice-app/platon_api/rest_api/search_engine/search_engine.py:
```diff
@@ -0,0 +1,295 @@
+        if request.GET.get("sorting_criteria",False):
+            # Take sorting criteria from URL
+            sotring_criteria = request.GET.get("sorting_criteria",False).replace('"',"")
+            if sotring_criteria not in searchEngine.sotring_criteria_list:
```
Done

HUmutOzdemir commented on practice-app/platon_api/rest_api/search_engine/search_engine.py:
```diff
@@ -0,0 +1,295 @@
+                return "Wrong input. Please give an appropriate input!!"
+        else:
+            # Default sorting criteria
+            sotring_criteria = ""
```
Done

HUmutOzdemir commented:
> I couldn't find an error other than simple spelling mistakes. Everything looks good. Well done

Thank you for your feedback. I changed all of my misspelled words.
